bindings/haskell/haskell-src/OpenDAL.hs

```diff
@@ -298,6 +312,30 @@ newOp scheme hashMap = do
                   errMsg <- peekCString (errorMessage ffiResult)
                   return $ Left $ OpenDALError code errMsg
 
+-- | Creates a new OpenDAL operator via `HashMap` with logger. The logger parameter is a function that accepts a string. OpenDAL will call this function to output logs.
+newOpWithLogger :: String -> HashMap String String -> LogLevel -> (String -> IO ()) -> IO (Either OpenDALError Operator)
```

Comment:
I reject this API because it is not extensible and overlaps with other potential APIs. OpenDAL aims to make all APIs plugable, allowing users to easily compose them as needed.

If we add `newOpWithLogger`, do we need to add `newOpWithTracing`, `newOpWithMetrics`? It doesn't make sense to our public API.

> By the way, `newOp` is also a bad name to me, I prefer to have `newOperator`. New users may don't know what's `Op` for.

I completely understand your motivation to make it easy for users to get started using opendal. However, I believe that this approach is incorrect. Instead, we should build and provide a clean and pluggable API that users can integrate with their own projects.

Reply:
How about the API in new commit? I move all config about creating operator into `OperatorConfig`, and instance `IsString` for it.

For default config, users only need to write `newOperator "memory"`. For advanced config, they can use haskell feature to extend default config, like 
`newOperator "memory" {ocLogConfig = Just $ OperatorLogConfig Debug logFn}`

As result, there is only `newOperator` using to create operator. If we want to add more feature about it, we just need to modify `OperatorConfig` and it won't make break change or add other APIs.